File: Scripter/QuickStarts/RunScriptMethods/ExternalClass.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using RunScriptMethods;

public class ExternalClass
{
    private readonly MainForm owner;

    internal ExternalClass(MainForm owner)
    {
        this.owner = owner;
    }

    public int X { get; set; } = 100;

    public int Y { get; set; } = 200;

    public string MainWindowText
    {
        get => owner.Text;
        set => owner.Text = value;
    }
}

File: Scripter/QuickStarts.VB/SimpleScriptRun/ExternalClass.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Public Class ExternalClass
    Private ReadOnly owner As Form1

    Friend Sub New(owner As Form1)
        Me.owner = owner
    End Sub

    Public Property X As Integer = 100

    Public Property Y As Integer = 200

    Public Property MainWindowText As String
        Get
            Return owner.Text
        End Get
        Set(value As String)
            owner.Text = value
        End Set
    End Property
End Class
